apps/web/src/lib/sections/BlogHighlights.svelte

```diff
@@ -0,0 +1,247 @@
+<script lang="ts">
+	import { formatDate } from '$lib/utils/formatDate';
+
+	// decalre the type of the posts
+	interface PostI {
+		feature_image: string;
+		published_at: string;
+		title: string;
+		url: string;
+		primary_author: {
+			name: string;
+		};
+		custom_excerpt: string;
+	}
+
+	export let posts: Array<PostI>;
```

Comment:
```suggestion
  const { posts }: PostI[] = $props();
```

Reply:
![CleanShot 2024-12-08 at 15 28 44@2x](https://github.com/user-attachments/assets/6ee4b5ac-37a5-46aa-a44f-0f3c1e9d74b0)

I have no idea what this means.